app/src/main/java/com/google/android/samples/socialite/ui/videoedit/VideoEditScreen.kt

```diff
@@ -91,37 +89,67 @@ fun VideoEditScreen(
     uri: String,
     onCloseButtonClicked: () -> Unit,
     navController: NavController,
+    viewModel: VideoEditScreenViewModel = hiltViewModel(),
 ) {
-    val context = LocalContext.current
-
-    val viewModel: VideoEditScreenViewModel = hiltViewModel()
     viewModel.setChatId(chatId)
-
-    val isFinishedEditing = viewModel.isFinishedEditing.collectAsStateWithLifecycle()
-    if (isFinishedEditing.value) {
-        navController.popBackStack("chat/$chatId", false)
-    }
-
-    val isProcessing = viewModel.isProcessing.collectAsState()
-
+    val isFinishedEditing by viewModel.isFinishedEditing.collectAsStateWithLifecycle()
+    val isProcessing by viewModel.isProcessing.collectAsStateWithLifecycle()
     var removeAudioEnabled by rememberSaveable { mutableStateOf(false) }
     var overlayText by rememberSaveable { mutableStateOf("") }
     var redOverlayTextEnabled by rememberSaveable { mutableStateOf(false) }
     var largeOverlayTextEnabled by rememberSaveable { mutableStateOf(false) }
+    val context = LocalContext.current
+    if (isFinishedEditing) {
+        navController.popBackStack("chat/$chatId", false)
+    }
```

Comment:
Also I this logic should be considered as side-effect caused by `isFinishedEditing`, so I think it should be isolated by `LaunchedEffect`